### PR TITLE
Add Release Doc

### PR DESCRIPTION
**Pull request by astoycos**
Another step towards #393

### CODE REVIEW

File: RELEASE.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
+# Release Process
+
+## Overview
+
+A release for the bpfd project is comprised of the following major components:
+
+- Built bpfd + bpfctl binaries
```
Comment (anfredette) [inline RELEASE.md:7]:
```suggestion
- bpfd + bpfctl binaries
```

Comment (anfredette) [reply RELEASE.md:7]: "Built" seems redundant.

File: RELEASE.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    API helpers.
+- Corresponding `bpfd-api`, `bpfd`, and `bpfctl` rust crates which house the rust client for the bpfd GRPC API
+- The following core component container images with tag <RELEASE_VERSION>:
+  - `quay.io/bpfd/bpfd`
+  - `quay.io/bpfd/bpfctl`
+  - `quay.io/bpfd/bpfd-operator`
+  - `quay.io/bpfd/bpfd-agent`
+  - `quay.io/bpfd/bpfd-operator-bundle`
+- The relevant example bytecode container images with tag <RELEASE_VERSION> who's source
+  code is located in the bpfd project:
```
Comment (anfredette) [inline RELEASE.md:26]:
```suggestion
- The relevant example bytecode container images with tag <RELEASE_VERSION> from source
  code located in the bpfd project:
```

File: RELEASE.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+  - `quay.io/bpfd-bytecode/go_xdp_counter`
+  - `quay.io/bpfd-bytecode/go_tc_counter`
+  - `quay.io/bpfd-bytecode/go_tracepoint_counter`
+- The relevant example userspace container images with tag <RELEASE_VERSION> who's source
+  code is located in the bpfd project:
```
Comment (anfredette) [inline RELEASE.md:31]:
```suggestion
- The relevant example userspace container images with tag <RELEASE_VERSION> from source
  code located in the bpfd project:
```

**Review by anfredette — COMMENTED** [review]
LGTM.  Just a few minor editorial suggestions that you can take or leave.

File: RELEASE.md (hunk 1 above)
Comment (astoycos) [reply RELEASE.md]: Done

**Review by anfredette — COMMENTED** [review]
LGTM (other than the nit)

File: RELEASE.md [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+  - `quay.io/bpfd/bpfd-operator`
+  - `quay.io/bpfd/bpfd-agent`
+  - `quay.io/bpfd/bpfd-operator-bundle`
+- The relevant example bytecode container images with tag <RELEASE_VERSION> from source
+  code located in the bpfd project:
+  - `quay.io/bpfd-bytecode/go_xdp_counter`
+  - `quay.io/bpfd-bytecode/go_tc_counter`
+  - `quay.io/bpfd-bytecode/go_tracepoint_counter`
+- The relevant example userspace container images with tag <RELEASE_VERSION> from source
+  code is located in the bpfd project:
```
[the remark this answers is not in the document]
Comment (astoycos) [reply RELEASE.md:31]: Done :)

**Comment by anfredette** [comment]
I guess we can remove the hold label since it's merged :-)

**Pull request merged**